lib/ipc_setup.c

```diff
@@ -486,8 +487,12 @@ int qb_ipcc_setup_connect_continue(struct qb_ipcc_connection *c, struct qb_ipc_c
 		qb_ipcc_us_sock_close(c->setup.u.us.sock);
 		return -ENOMEM;
 	}
-
+retry:
 	res = qb_ipc_us_recv_msghdr(data);
+	if (res == -EAGAIN && ++retry_count < 10) {
+		usleep(100000);
```

Comment:
FYI usleep() is considered obsolete, though it's generally available everywhere of interest. Pacemaker defines pcmk__sleep_ms() to wrap the various possibilities depending on what's available. These days you could probably just assume nanosleep() is present ...